### DISCUSSION TITLE
Which is the best approach to deploy MERN stack app on AMAZON EC2?

### DISCUSSION THREAD

**Discussion by chandnie** [community/community#20670] [Discussions] [answered]
I am googling a lot to be sure of the best approach to deploy the MERN stack app on AWS ec2 ... in some examples, Nginx is being used in the server (expressjs) part and in some cases, it's being used for the Recatjs part, and the express js part is just hosted in node, React even can be hosted in s3 I guess .. so what is the best approach, however?

**Comment by chiragmehta900** [comment, marked as answer]
Without configuring the server yourself in EC2, you can simply use managed Node.js hosting provided by AWS Elastic Beanstalk.

If you want to directly configure EC2 instance and deploy, you can follow this tutorial to do the deployment 

`Visit to this link for more update.`
https://jasonwatmore.com/post/2019/11/18/react-nodejs-on-aws-how-to-deploy-a-mern-stack-app-to-amazon-ec2

**Comment by malwarebo** [comment]
You have many options are:
1. Standalone EC2
2. Elastic Beanstalk
3. Docker 

While 2 and 3 are add a little complexity in handling things over the long run, I guess option 1 should be fine for you.
I don't know what your web app architecture is but based on your question I think 1 is simple and easy enough to get you up.
